### ISSUE TITLE
Publish the keyring provider in this homebrew tap

### ISSUE THREAD

**Issue by izgeri** [cyberark/homebrew-tools#33]
## Is your feature request related to a problem? Please describe.

I would like to install Summon / Summon Keyring provider using homebrew, but the keyring provider isn't available here, which means I can install Summon here but I have to follow a [manual process with multiple steps](https://github.com/cyberark/summon-keyring#install) for installing the keyring provider.

## Describe the solution you would like

Add the Summon Keyring provider to this tap - I'm not 100% sure what the UX would be (you still need to `pip install keyring` I'd guess), but maybe it would be better?

**Comment by doodlesbykumbi** [comment]
@izgeri I agree that a formula for the keyring provider would be a lot more convenient.

I had a go at putting together a very bare bones one. Here's how it looks:
```
class SummonKeyring < Formula
  desc "Cross-platform provider for Summon that talks to keyrings"
  homepage "https://github.com/cyberark/summon-keyring"
  version "0.2.1"
  bottle :unneeded
  url "https://raw.githubusercontent.com/cyberark/summon-keyring/v0.2.1/ring.py"
  sha256 "4df95dff663e43fc996f6e876b3baa7016b839eea2ab5eacbcdf9796c934992d"

  def install
    target = lib/"summon"
    target.install "ring.py" => "summon-keyring"
  end
end
```

This seems to do most of what we might want from a formula. Some notes:

1. We're able to grab the raw script from GH so perhaps that circumvents the desire to add the script into the release.
2. I noticed that as part of your README update you mentioned getting this error `AttributeError: module 'keyring' has no attribute 'get_password'`. It turns out that happens entirely because we chose to rename the provider to `keyring.py` when using it. If you call it anything other than that then the error never happens. Symlinking to `ring.py` has the same effect because I suppose when Python introspects it sees that the original file is called `ring.py`, had we been running `import ring` instead of `import keyring` then symlinking would have no effect.
3. This formula doesn't handle the installation of the `keyring` python dependency. However, Homebrew definitely has a mechanism for installing dependencies as part of the formula, e.g. using [resources](https://docs.brew.sh/Python-for-Formula-Authors#example), but then things get complicated because we don't know the Python version on the users machine. A different version of the keyring module would be needed depending on the Python version. 
4. From (3) above, the provider is only meant to support Python 3. I'm not sure how to enforce that when installing the formula. It could be as simple as checking what `python -v` returns. But then we probably need to be flexible enough to account for the fact that Python 3 might only be accessible via `python3` not `python`. I'm not sure how to do this but it would be ideal to be able to specify on `brew install` the path to the python binary to use to execute the provider.
   It's also possible to add the Python 3 brew formula as a dependency but then what about people who have Python 3 already installed, or those who don't want to install Python via homebrew.
5. From (4), right now the provider script has the shebang `#!/usr/bin/env python` which means it uses whichever python is available. We will probably want to add something like this to the provider script
```
import sys

if sys.version_info[0] < 3:
    raise Exception("Python 3 or a more recent version is required.")
```

Resolving (3), (4) and (5) would make the UX a lot nicer.

**Comment by izgeri** [comment]
Thanks @doodlesbykumbi. It sounds like it's not a simple solution - if I can think of anything I can do to help progress this, I'll let you know.
